packages/server-node/src/features/model/request-model-action-handler.ts

```diff
@@ -41,23 +41,21 @@ export class RequestModelActionHandler implements ActionHandler {
     @inject(ModelSubmissionHandler)
     protected submissionHandler: ModelSubmissionHandler;
 
-    async execute(action: Action): Promise<Action[]> {
+    async execute(action: RequestModelAction): Promise<Action[]> {
         this.logger.debug('Execute RequestModelAction:', action);
-        if (isRequestModelAction(action)) {
-            this.modelState.setAll(action.options ?? {});
+        this.modelState.setAll(action.options ?? {});
 
-            this.notifyClient('Model loading in progress');
-            await this.sourceModelStorage.loadSourceModel(action);
-            this.notifyClient();
-        }
+        this.notifyClient('Model loading in progress');
+        await this.sourceModelStorage.loadSourceModel(action);
+        this.notifyClient();
         return this.submissionHandler.submitModel();
     }
 
     protected notifyClient(message?: string): void {
         const severity = message ? 'INFO' : 'NONE';
         this.actionDispatcher.dispatchAll(
-            new GLSPServerStatusAction(severity, message ?? ''),
-            new ServerMessageAction(severity, message ?? '')
+            ServerMessageAction.create(message ?? '', { severity }),
+            ServerMessageAction.create(message ?? '', { severity })
```

Comment:
```suggestion
            ServerMessageAction.create(message ?? '', { severity }),
            ServerMessageAction.create(message ?? '', { severity })
```

Any reason why we need to dispatch the same message twice?

Reply:
Could catch, one of them should be a `ServerStatusAction`